File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeWeightMessage.scala





















// Generated on 05/03/2013 10:34:56
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeWeightMessage extends MessageDeserializer {
  val messageId = 5793

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ExchangeWeightMessage.messageId

    val currentWeight: Int
    val maxWeight: Int

    def serialize(buf: Buffer) {
      buf.writeInt(currentWeight)
      buf.writeInt(maxWeight)
    }

  }

  protected def create(currentWeight: Int, maxWeight: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val currentWeight = buf.readInt()
    val maxWeight = buf.readInt()

    create(currentWeight, maxWeight)
  }

}

object ExchangeWeightMessage extends ExchangeWeightMessage {
  case class MessageImpl(currentWeight: Int, maxWeight: Int) extends Message
  type Target = MessageImpl

  override def create(currentWeight: Int, maxWeight: Int) = MessageImpl(currentWeight, maxWeight)

  def apply(currentWeight: Int, maxWeight: Int) = create(currentWeight, maxWeight)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeWeightMessage.java


// Generated on 05/08/2013 19:37:55
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeWeightMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5793;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int currentWeight;
    public int maxWeight;
    
    public ExchangeWeightMessage() { }
    
    public ExchangeWeightMessage(int currentWeight, int maxWeight) {
        this.currentWeight = currentWeight;
        this.maxWeight = maxWeight;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(currentWeight);
        buf.writeInt(maxWeight);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        currentWeight = buf.readInt();
        if (currentWeight < 0)
            throw new RuntimeException("Forbidden value on currentWeight = " + currentWeight + ", it doesn't respect the following condition : currentWeight < 0");
        maxWeight = buf.readInt();
        if (maxWeight < 0)
            throw new RuntimeException("Forbidden value on maxWeight = " + maxWeight + ", it doesn't respect the following condition : maxWeight < 0");
    }
    
}
